pkg/infoschema/sieve.go

```diff
@@ -75,6 +75,17 @@ func (s *Sieve[K, V]) SetCapacity(capacity uint64) {
 	s.capacity = capacity
 }
 
+func (s *Sieve[K, V]) SetCapacityAndWaitEvict(capacity uint64) {
```

Comment:
I'm not sure if there will be two call, one wants to increase capacity and the other wants to decrease. Maybe we should add another lock to make sure `SetCapacityAndWaitEvict` is not concurrently executing.

Reply:
Do you mean the access of `s.size` and `s.capacity` is not protected by lock?